turtlebot3_example/turtlebot3_example/turtlebot3_position_control/turtlebot3_position_control.py

```diff
@@ -0,0 +1,180 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors: Ryan Shim, Gilbert
+
+import math
+import numpy
+from rclpy.node import Node
+from rclpy.qos import QoSProfile
+import sys
+import termios
+from turtlebot3_example.turtlebot3_position_control.turtlebot3_path import Turtlebot3Path
+
+from geometry_msgs.msg import Twist
+from nav_msgs.msg import Odometry
+
+terminal_msg = """
+Turtlebot3 Position Control
+------------------------------------------------------
+From the current pose,
+x: goal position x (unit: m)
+y: goal position y (unit: m)
+theta: goal orientation (range: -180 ~ 180, unit: deg)
+------------------------------------------------------
+"""
+
+
+class Turtlebot3PositionControl(Node):
+
+    def __init__(self):
+        super().__init__('turtlebot3_position_control')
+
+        """************************************************************
+        ** Initialise variables
+        ************************************************************"""
+        self.odom = Odometry()
+        self.last_pose_x = 0.0
+        self.last_pose_y = 0.0
+        self.last_pose_theta = 0.0
+        self.goal_pose_x = 0.0
+        self.goal_pose_y = 0.0
+        self.goal_pose_theta = 0.0
+        self.step = 1
+        self.get_key_state = False
+        self.init_odom_state = False  # To get the initial pose at the beginning
+
+        """************************************************************
+        ** Initialise ROS publishers and subscribers
+        ************************************************************"""
+        qos = QoSProfile(depth=10)
+
+        # Initialise publishers
+        self.cmd_vel_pub = self.create_publisher(Twist, 'cmd_vel', qos)
+
+        # Initialise subscribers
+        self.odom_sub = self.create_subscription(
+            Odometry,
+            'odom',
+            self.odom_callback,
+            qos)
+
+        """************************************************************
+        ** Initialise timers
+        ************************************************************"""
+        self.update_timer = self.create_timer(0.010, self.update_callback)  # unit: s
+
+        self.get_logger().info("Turtlebot3 position control node has been initialised.")
+
+    """*******************************************************************************
+    ** Callback functions and relevant functions
+    *******************************************************************************"""
+    def odom_callback(self, msg):
+        self.last_pose_x = msg.pose.pose.position.x
+        self.last_pose_y = msg.pose.pose.position.y
+        _, _, self.last_pose_theta = self.euler_from_quaternion(msg.pose.pose.orientation)
```

Comment:
What is `_,` this symbol??

Reply:
I refered to [this](https://mingrammer.com/underscore-in-python/). Let me know if you know any other better expressions.